R/cube-residuals.R

```diff
@@ -0,0 +1,141 @@
+# standardized residuals formed by the subvariable-wise margins
+# the internal chisq tests use rowSums and colSums, but we need
+# the more abstract 'margin' (specifically cubeMarginTable) to
+# get the right numbers for multiple response.
+standardizedMRResiduals <- function(cube, types){
+    # grab the dims from as.array because sometimes cat x MR has one too many
```

Comment:
Really? Why?

Reply:
poorly worded. the cube itself has more dims than `as.array(cube)` does. The desired ndims is that of the array

Reply:
_You may be interested to know_ that the `dim()` method on `CrunchCube` handles this for you: https://github.com/Crunch-io/rcrunch/blob/master/tests/testthat/test-cube-mr.R#L13-L23

I'll verify and clean this up here.